### PR TITLE
fixes #354: Allow text-search/indexing to be disabled.

### LINKED ISSUE #354: Allow text search to be disabled
Text search is based on Lucene indices. These indices take resources to generate. That is typically a negligible amount, but for very large amounts of data, this can add up (eg: Ignite's 10.5 million records take about 7 minutes to be fully indexed).

If text search is not used, then this is a waste of resources. A configuration option should be added to disable this functionality.

### PR DESCRIPTION
**Pull request by guusdk**
This introduces a new system property, `conversation.search.index-enabled`, that, when set to 'false', disables the full-text search functionality and corresponding indexing tasks.

Disabling this functionality can save some resources on instances that do not use the feature, and have a large amount of messages in their database.

### CODE REVIEW

File: src/java/org/jivesoftware/openfire/index/LuceneIndexer.java [hunk 1]
```diff
@@ -409,6 +439,11 @@ public Future<Integer> getIndexRebuildProgress()
      */
     public synchronized IndexSearcher getSearcher() throws IOException
     {
+        if (!ENABLED.getValue()) {
+            Log.debug("Unable to get index: indexing is disable by configuration.");
```
Comment (Fishbowler) [inline src/java/org/jivesoftware/openfire/index/LuceneIndexer.java:443]:
```suggestion
            Log.debug("Unable to get index: indexing is disabled by configuration.");
```

File: src/i18n/monitoringservice_i18n.properties [hunk 2]
```diff
@@ -22,10 +22,11 @@ system_property.conversation.maxTime=The max number of minutes a conversation ca
 system_property.conversation.messageArchiving=Enable archiving of one-to-one messages
 system_property.conversation.metadataArchiving=Enable archiving of conversation metadata
 system_property.conversation.roomArchiving=Enable archiving of MUC messages
-system_property.conversation.roomArchiving.PMinPersonalArchive=Controls whether MUC PMs are stored in the Personal Archive for the purposes of Message Arrchiving (XEP-0313)
-system_property.conversation.roomArchiving.PMinRoomArchive=Controls whether MUC PMs are stored in the Room Archive for the purposes of Message Arrchiving (XEP-0313)
```
[the remark this answers is not in the document]
Comment (guusdk) [reply src/i18n/monitoringservice_i18n.properties:26]: Yes, it is sad to see those go.

File: src/java/org/jivesoftware/openfire/index/LuceneIndexer.java (hunk 1 above)
Comment (guusdk) [reply src/java/org/jivesoftware/openfire/index/LuceneIndexer.java]: Ah, I replaced 4 of these in a commit that I just pushed.

**Comment by guusdk** [comment]
Rebased, and performed basic testing.

**Pull request merged**